File: backend/src/main/java/com/huertohogar/model/entity/StoreLocation.java
package com.huertohogar.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "store_locations")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class StoreLocation {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    
    @Column(nullable = false, length = 200)
    private String name;
    
    @Column(nullable = false, length = 500)
    private String address;
    
    @Column(nullable = false)
    private Double latitude;
    
    @Column(nullable = false)
    private Double longitude;
    
    @Column(length = 20)
    private String phone;
    
    @Column(length = 200)
    private String email;
    
    @Column(name = "opening_hours", length = 200)
    private String openingHours;
}

File: app/src/main/java/com/huertohogar/data/model/StoreLocation.kt
package com.huertohogar.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Nuestro modelo de ubicaciones de las tiendas para el mapa
 */
@Entity(tableName = "store_locations")
data class StoreLocation(
    @PrimaryKey
    val id: String,
    val name: String,
    val city: String,
    val address: String,
    val latitude: Double,
    val longitude: Double,
    val phone: String = "",
    val hours: String = "" // Horario de atención
)
